tests/mixins/test_uploads.py

```diff
@@ -47,6 +49,41 @@ def test_upload_song(self, config, yt_auth):
         response = yt_auth.upload_song(get_resource(config["uploads"]["file"]))
         assert response.status_code == 409
 
+    def test_upload_song_and_verify(self, config, yt_auth: YTMusic):
+        """Upload a song and verify it can be retrieved after it finishes processing."""
+        upload_response = yt_auth.upload_song(get_resource(config["uploads"]["file"]))
+        if not isinstance(upload_response, str) and upload_response.status_code == 409:
+            # Song is already in uploads. Delete it and re-upload
+            songs = yt_auth.get_library_upload_songs()
+            delete_response = None
+            for song in songs:
+                if song.get("title") in config["uploads"]["file"]:
+                    delete_response = yt_auth.delete_upload_entity(song["entityId"])
+            assert delete_response == "STATUS_SUCCEEDED"
+            # Need to wait for song to be fully deleted
+            time.sleep(10)
```

Comment:
I didn't like adding a hard sleep of 10 seconds here but it turns out that even though the delete response was "success" and a subsequent call to `get_library_uploads()` would yield an empty list, the next `upload_song()` call was still resulting in a 409 Conflict. 10 seconds seems to consistently be enough to ensure that it's fully deleted.

Reply:
Fine with me, there are some similar issue with the playlist end2end returning conflicts. Better than a flaky test